packages/dito/app/routes/vorpruefung.ergebnis/ResultPage.tsx

```diff
@@ -0,0 +1,232 @@
+import Box from "@digitalcheck/shared/components/Box.tsx";
+import Container from "@digitalcheck/shared/components/Container";
+import Heading from "@digitalcheck/shared/components/Heading.tsx";
+import { NumberedList } from "@digitalcheck/shared/components/List";
+import RichText from "@digitalcheck/shared/components/RichText.tsx";
+import {
+  ControlPointOutlined,
+  HelpOutline,
+  RemoveCircleOutline,
+} from "@digitalservicebund/icons";
+import classNames from "classnames";
+import { preCheck } from "resources/content";
+import {
+  PreCheckAnswers,
+  TQuestion,
+} from "routes/vorpruefung.$questionId/route";
+import Accordion from "../../components/Accordion.tsx";
+import ResultForm from "./ResultForm.tsx";
+import ResultHeader from "./ResultHeader";
+import { ResultType, TResult } from "./TResult.tsx";
+
+const { questions } = preCheck;
+
+type QuestionAndAnswer = {
+  question: TQuestion;
+  answer: string;
+};
+
+const title = {
+  interoperability: {
+    [ResultType.POSITIVE]: preCheck.result.interoperability.positive.title,
+    [ResultType.NEGATIVE]: preCheck.result.interoperability.negative.title,
+    [ResultType.UNSURE]: preCheck.result.interoperability.unsure.title,
+  },
+  digital: {
+    [ResultType.POSITIVE]: preCheck.result.positive.title,
+    [ResultType.NEGATIVE]: preCheck.result.negative.title,
+    [ResultType.UNSURE]: preCheck.result.unsure.title,
+  },
+};
+
+const nextSteps = {
+  [ResultType.POSITIVE]: preCheck.result.positive.nextSteps,
+  [ResultType.NEGATIVE]: preCheck.result.negative.nextSteps,
+};
+
+function matchQuestionsAndAnswers(answers: PreCheckAnswers) {
+  return Object.keys(answers).map((key) => {
+    const question = questions.find((q) => q.id === key);
+    if (!question) {
+      throw new Error(`No matching question found for key: ${key}`);
+    }
+    return { question, answer: answers[key] };
+  });
+}
+
+export default function ResultPage({
+  answers,
+  result,
+}: Readonly<{
+  answers: PreCheckAnswers;
+  result: TResult;
+}>) {
+  const questionsAndAnswers = matchQuestionsAndAnswers(answers);
+
+  const questionsForDigitalAndSure = questionsAndAnswers.filter(
+    (tuple) => !tuple.question.interoperability && tuple.answer !== "unsure",
+  );
+  const questionsForDigitalAndUnsure = questionsAndAnswers.filter(
+    (tuple) => !tuple.question.interoperability && tuple.answer === "unsure",
+  );
+  const questionsForInteroperabilitySure = questionsAndAnswers.filter(
+    (tuple) => tuple.question.interoperability && tuple.answer !== "unsure",
+  );
+  const questionsForInteroperabilityUnsure = questionsAndAnswers.filter(
+    (tuple) => tuple.question.interoperability && tuple.answer === "unsure",
+  );
+
+  function getReasonListItem(data: {
```

Comment:
As this and the next three functions don't access local variables, I would pull them out of `ResultPage` and also simplify them to something like 

```
function getReasonListItem(tuple: QuestionAndAnswer) {
  const classes = classNames("w-28 h-auto", {
    "fill-green-900": tuple.answer === "yes",
    "fill-red-900": tuple.answer === "no",
  });
  const resultType =
    tuple.answer === "no" ? "negativeResult" : "positiveResult";
  const resultHint = tuple.question?.resultHint?.[resultType];
  return (
    <>
      {tuple.answer === "yes" && <ControlPointOutlined className={classes} />}
      {tuple.answer === "no" && <RemoveCircleOutline className={classes} />}
      {tuple.answer === "unsure" && <HelpOutline className={classes} />}
      <span>
        <span>
          {tuple.question?.[resultType]}
          {resultHint && <RichText markdown={resultHint} />}
        </span>
      </span>
    </>
  );
}

function getReasonsList(questions: QuestionAndAnswer[], intro: string) {
  return (
    questions.length !== 0 && (
      <>
        <RichText className="mt-40" markdown={intro} />
        <ul className="ds-stack-16 mt-16">
          {questions
            .toSorted((t1) => (t1.answer === "yes" ? -1 : 1))
            .map((tuple) => (
              <li key={tuple.question.id} className="flex items-start gap-12">
                {getReasonListItem(tuple)}
              </li>
            ))}
        </ul>
      </>
    )
  );
}
``` 
> [!WARNING]
> Not tested!